Add PROJECT template

polynomial opened the pull request:
The goal of the project template is to define a method for engineers
that are researching, implementing/modifying, testing, and maintaining
a system.

mbbx6spp commented on docs/PROJECT.asciidoc:
```diff
@@ -0,0 +1,74 @@
+=== Project Definition:
+
+* What is the intended benefit of this project
+* What are the success criteria
```
Much better. I might just clean it up a little more like so though:

```asciidoc
Answer the following questions in this section:
* What is the intended benefit of this project *for the business*?
* What are the criteria to use to evaluate project success and at which stages/phases of the project do they apply?
```

mbbx6spp commented on docs/PROJECT.asciidoc:
```diff
@@ -0,0 +1,74 @@
+
+=== Definitions:
+
+* Define terms that will be used in this document for better clarity.
```
Provide examples in Asciidoc markup to make the barrier to doing this lower :)

mbbx6spp commented on docs/PROJECT.asciidoc:
```diff
@@ -0,0 +1,74 @@
+* Define terms that will be used in this document for better clarity.
+
+=== Business Impact:
+
```
Add a leading sentence to the list of something like the following `Consider the answers to the following questions while writing this section:`.

mbbx6spp commented on docs/PROJECT.asciidoc:
```diff
@@ -0,0 +1,74 @@
+
+=== Business Impact:
+
+* How will this benefit the business
```
You are missing question marks everywhere :)

mbbx6spp commented on docs/PROJECT.asciidoc:
```diff
@@ -0,0 +1,74 @@
+=== Business Impact:
+
+* How will this benefit the business
+* How does this align with current key objectives
```
objectives or business initiatives? Can it be aligned up via BetterWorks goals or something (for internal use)?

mbbx6spp commented on docs/PROJECT.asciidoc:
```diff
@@ -0,0 +1,74 @@
+=== Inputs:
+
+* Identify key stakeholder groups and detail why they must be included
+* Identify key requirements and detail should vs must have
```
Provide an example of this. For internal use we could retrofit the format of the secret store requirements Gist?

mbbx6spp commented on docs/PROJECT.asciidoc:
```diff
@@ -0,0 +1,74 @@
+* Identify key stakeholder groups and detail why they must be included
+* Identify key requirements and detail should vs must have
+
+=== Theory:
```
This section needs to be retitled for business projects. This is not reframed at all. I can work on that AFTER channels! ;)

mbbx6spp commented on docs/PROJECT.asciidoc:
```diff
@@ -0,0 +1,74 @@
+=== Related Work:
+
+* Identify similar tasks the business has undertaken and how what was learned
+  there that may apply here.
```
tasks or projects or initiatives? Tasks implies something much finer grained.

mbbx6spp commented on docs/PROJECT.asciidoc:
```diff
@@ -0,0 +1,74 @@
+  there that may apply here.
+* Research other companies undertaking a similar project and include their
+  outcomes here. This should include alternative approaches (different
+  technologies, team organization, skills, etc).
```
Even citing research/industry technical papers (and explaining why they are relevant) that are highly related to the project's domains should be included here. Or perhaps this should be included in the currently titled "Theory" section? I think the "Theory" section should contain references to well understood technical material and this should reference related projects/studies/research.

mbbx6spp commented on docs/PROJECT.asciidoc:
```diff
@@ -0,0 +1,74 @@
+* Identify key milestones for the project
+* Scope work for each of these milestones
+* Identify project dependencies between milestones
+* Document key technical decisions 
```
Also add in what criteria assessments occur and when. What do they tell us at each phase/stage?

mbbx6spp commented on docs/PROJECT.asciidoc:
```diff
@@ -0,0 +1,74 @@
+* Identify project dependencies between milestones
+* Document key technical decisions 
+
+=== Results (Data):
```
The content of this section needs to be a little more reframed in terms of the business project mindset.

mbbx6spp commented on docs/PROJECT.asciidoc:
```diff
@@ -0,0 +1,74 @@
+* Collect data around scope modifications
+* Collect and present data around task timing and dependency identification
+  success.
+* Link to long term measurements of business impact for the project (good/bad)
```
This part is good but MOAR please. Perhaps examples?

mbbx6spp commented on docs/PROJECT.asciidoc:
```diff
@@ -0,0 +1,74 @@
+  task here is to help the team improve not to place blame.
+* Identify what may have been difficult or impossible to measure.
+
+=== Future Work:
```
Section needs reframing for business project perspective. Too science experimenty.

mbbx6spp commented on docs/README.asciidoc:
```diff
@@ -27,6 +27,10 @@ understanding, and vision:
   performance tuning, and other activities that require more scope and
   definition than the typical Jira ticket from a technical perspective.
+* link:PROJECT.asciidoc[] - this document uses the link:EXPERIMENT.asciidoc[]
+  template for bigger projects. This should be used when you need to
```
"Bigger" is subjective. I would argue the purpose of this document is for longer running initiatives with more people and/or teams involved rather than "bigger".

mbbx6spp commented:
Vast improvement. I would love to see more examples though. I think that would make it far more useful for consumers of the document.

polynomial commented:
Thanks for the feedback, I believe I addressed all those comments. There is a TODO regarding an example for stakeholders. I'd like to take the internal version we have and sanitize it for inclusion into this repository. Suggestions? @mbbx6spp

mbbx6spp commented:
Only looked at this on my phone but only question I have so far is whether inputs section is just a link to the requirements doc?

polynomial commented:
Ok, I linked the requirements doc too. Let me know if there is further stuff needed on this PR @mbbx6spp

mbbx6spp commented:
Could you rebase one more time against just master? Sorry, GH does the wrong thing here in the diffs and I get nervous seeing files that are already on master in the diff of a PR. Shame on GH.

polynomial commented:
Yes m'lady.
